blogs/2025/05/12/agent-mode-meets-mcp.md

```diff
@@ -0,0 +1,62 @@
+
+[**Dynamic Tool Discovery**](https://modelcontextprotocol.io/docs/concepts/tools#tool-discovery-and-updates) _(list-changed events )_ is a concept that allows servers to notify clients that their available tools changed. Without dynamic tool discovery, the list of tools a server provides is defined once during initialization. Making tools more dynamic and adaptive can avoid overwhelming AI agent with irrelevant tools. MCP servers can dynamically add/remove tools based on scope, context or state—like a to-do server that updates tools as tasks are completed or an API monitoring server showing tools depending on which features are enabled for an endpoint. GitHub’s official MCP server is using this in their [“dynamic toolset discovery”](https://github.com/github/github-mcp-server/?tab=readme-ov-file#dynamic-tool-discovery) feature, which gives client control to enable/disable sets of tools on demand during an agent session.
+
+[**Tool annotations**](https://modelcontextprotocol.io/docs/concepts/tools#tool-annotations) are a recent addition to the spec and are meant for clients like VS Code as hints on how to display and manage tools. VS Code added support for human-readable tool names (_title_). The support for _readOnlyHint_ flag allows tools to run similar to built-in read-only tools.
+
+[**Streamable HTTP**](https://modelcontextprotocol.io/specification/2025-03-26/basic/transports#streamable-http) is the latest remote transport mechanism for MCP, simplifying the development and deployment of MCP servers, particularly in cloud and serverless environments. It replaces using Server-Sent Events (SSE) as a more efficient and scalable remote transport. By having VS Code support for the latest spec, developers can stay current. Local MCP servers remain a great choice for developer tools, as they run on the local machine and have access to the workspace (via roots).
+
+As more servers adopt these primitives, agentic workflows can become more precise, contextual, and composable.
+
+## What's Next
+
+MCP has gained traction with official servers from GitHub, Playwright, Azure, and Perplexity, alongside a growing community of over a thousand servers. As more agents adopt the protocol, MCP is becoming foundational infrastructure for practical AI-assisted development. As the protocol matures and the ecosystem expands, we'll continue refining the experience based on your feedback and in collaboration with the MCP community.
```

Comment:
We could hint at planned features: Auth, Prompts, Resources and Registry. But it also gets stale quicker.